python/src/aiconfig/editor/client/src/components/AIConfigEditor.tsx

```diff
@@ -931,134 +935,143 @@ export default function EditorContainer({
-                  onUpdateParameters={onUpdatePromptParameters}
-                  defaultConfigModelName={aiconfigState.metadata.default_model}
-                  isRunButtonDisabled={isAnotherPromptRunning}
+                  addPrompt={(model: string) => onAddPrompt(0, model)}
                 />
-              </Flex>
-              {!readOnly && (
-                <div className={classes.addPromptRow}>
-                  <AddPromptButton
-                    getModels={callbacks?.getModels}
-                    addPrompt={(model: string) =>
-                      onAddPrompt(
-                        i + 1 /* insert below current prompt index */,
-                        model
-                      )
-                    }
-                  />
-                </div>
-              )}
-            </Stack>
-          );
-        })}
-      </Container>
-    </AIConfigContext.Provider>
+              </div>
+            )}
+            {aiconfigState.prompts.map((prompt: ClientPrompt, i: number) => {
+              const isAnotherPromptRunning =
+                runningPromptId !== undefined &&
+                runningPromptId !== prompt._ui.id;
+              return (
+                <Stack key={prompt._ui.id}>
+                  <Flex mt="md">
+                    <PromptMenuButton
+                      promptId={prompt._ui.id}
+                      onDeletePrompt={() => onDeletePrompt(prompt._ui.id)}
+                    />
+                    <PromptContainer
+                      prompt={prompt}
+                      getModels={callbacks?.getModels}
+                      onChangePromptInput={onChangePromptInput}
+                      onChangePromptName={onChangePromptName}
+                      cancel={callbacks?.cancel}
+                      onRunPrompt={onRunPrompt}
+                      onUpdateModel={onUpdatePromptModel}
+                      onUpdateModelSettings={onUpdatePromptModelSettings}
+                      onUpdateParameters={onUpdatePromptParameters}
+                      defaultConfigModelName={
+                        aiconfigState.metadata.default_model
+                      }
+                      isRunButtonDisabled={isAnotherPromptRunning}
+                    />
+                  </Flex>
+                  {!readOnly && (
+                    <div className={classes.addPromptRow}>
+                      <AddPromptButton
+                        getModels={callbacks?.getModels}
+                        addPrompt={(model: string) =>
+                          onAddPrompt(
+                            i + 1 /* insert below current prompt index */,
+                            model
+                          )
+                        }
+                      />
+                    </div>
+                  )}
+                </Stack>
+              );
+            })}
+          </Container>
+        </div>
+      </AIConfigContext.Provider>
+    </AIConfigEditorThemeProvider>
```

Comment:
question for myself to understand react/mantine: I'm looking at `RunPromptButton.tsx`, how does it pull the styling/theme from the relevant theme.ts for itself?

Reply:
nvm I see that the button is defined as 
```
    <Button
      onClick={onClick}
      disabled={disabledOrReadOnly}
      p="xs"
      size="xs"
      className="runPromptButton"
    >
    ```
    
    and the theme.ts has runPromptButton defined

Reply:
The theme/styles would be applied by:
- any element rendered by RunPromptButton which has a class name will have the style applied from the global styles set by theme (e.g. runPromptButton class)
- within mantine core code, the Button, Flex, Loader, Tooltip components used within RunPromptButton will themselves have styles applied based on applicable values set in context from the closest MantineThemeProvider